core/classifiers/classifier_utils.py

```diff
@@ -60,15 +59,92 @@ def extract_svm_parameters(clf):
     }
 
 
-def unicode_validator_for_classifier_data(var):
+def unicode_validator_for_classifier_data(classifier_data):
     """Validates that incoming object contains unicode literal strings."""
-    if isinstance(var, dict):
-        for k in var.keys():
+    if isinstance(classifier_data, dict):
+        for k in classifier_data.keys():
             if isinstance(k, str):
                 raise Exception('Expected %s to be unicode but found str.' % k)
-            unicode_validator_for_classifier_data(var[k])
-    if isinstance(var, (list, set, tuple)):
-        for item in var:
+            unicode_validator_for_classifier_data(classifier_data[k])
+    if isinstance(classifier_data, (list, set, tuple)):
+        for item in classifier_data:
             unicode_validator_for_classifier_data(item)
-    if isinstance(var, str):
-        raise Exception('Expected \'%s\' to be unicode but found str.' % var)
+    if isinstance(classifier_data, str):
+        raise Exception(
+            'Expected \'%s\' to be unicode but found str.' % classifier_data)
+
+
+FLOAT_INDICATOR_KEY = 'float_values'
+
+
+# pylint: disable=too-many-branches
+def convert_float_numbers_to_string_in_classifier_data(classifier_data):
```

Comment:
Hi @seanlip, so this is the final version. So, after completely implementing `string_values` algorithm, I realized that it can be easily modified to `float_values` where each subdict is augmented with a `float_values` key which contains a list of keys which should undergo the transformation.

Reply:
Sounds good!